crates/autopilot/src/run.rs

```diff
@@ -449,8 +449,14 @@ pub async fn run(args: Arguments) {
         None
     };
 
+    let persistence =
+        infra::persistence::Persistence::new(args.s3.into().unwrap(), Arc::new(db.clone())).await;
     let on_settlement_event_updater =
-        crate::on_settlement_event_updater::OnSettlementEventUpdater::new(eth.clone(), db.clone());
+        crate::on_settlement_event_updater::OnSettlementEventUpdater::new(
+            eth.clone(),
+            db.clone(),
```

Comment:
Would be nice to get rid of this db dependency now that we depend on persistence. Maybe persistence should have an inner object which exposes the same functions it does but will batch all calls in a single db transaction (and it can be awaited on to commit all changes before it's dropped)

Reply:
db still needed for `get_settlement_without_auction`, and moving this function to persistence can be a PR on it's own. 

AFAIS we have 3 thing that have to be done sequentially (with regards to the db communication):
1. get_settlement_without_auction to get the tx hash
2.get all auction data for the auction that corresponds to that tx hash
3. optionally write Observation data to the db.

Still not decided, but I am also ok with moving these calls into a single function under persistence (that would use a single db transaction), but that function has to use domain objects to 
1. recover settlement and build proper errors if needed
2. do the checks (verify winner, score, etc) in a domain and build proper errors if needed